fero/analysis.py

```diff
@@ -291,9 +359,54 @@ def make_prediction(
         self, prediction_data: Union[pd.DataFrame, List[dict]]
     ) -> Union[pd.DataFrame, List[dict]]:
         """Makes a prediction from the provided data using the most recent trained model for the analysis.
+        This method is optimized for analyses that support fast, bulk prediction. For analyses that do not support
+        bulk prediction, use `make_prediction_legacy`.
+
+        `make_prediction` takes either a data frame or list of dictionaries of values that will be sent to Fero
+        to make a prediction of what the targets of the Analysis will be. The results are returned as either a dataframe
+        or list of dictionaries with both the original prediction data and the predicted targets in each row or dict.
+        Each target has a `high`, `low`, and `mid` value and these are added to the target variable name with an `_`.
+
+        :param prediction_data:  Either a data frame or list of dictionaries specifying values to be used in the model.
+        :type prediction_data: Union[pd.DataFrame, List[dict]]
+        :raises FeroError: Raised if no model has been trained or the server returns an error message
+        :return: A data frame or list of dictionaries depending on how the function was called
+        :rtype: Union[pd.DataFrame, List[dict]]
+        """
+        if not self.has_trained_model:
+            raise FeroError("No model has been trained on this analysis.")
+
+        is_dict_list = isinstance(prediction_data, list)
+
+        prediction_df = (
+            pd.DataFrame(prediction_data) if is_dict_list else prediction_data
+        )
+
+        data_file = io.StringIO(json.dumps(prediction_df.to_dict(orient="split")))
+        upload_identifier = str(uuid.uuid4())
+        workspace_id = self._upload_file(
+            data_file, upload_identifier, BULK_PREDICTION_TYPE
+        )
+        prediction = self._poll_workspace_for_prediction(workspace_id)
```

Comment:
Might be worth doing this optionally and letting the user handle the async however they want to rather than blocking automatically.

Reply:
Todd/Alp were saying we should just do this for now -- happy to change, though